sagemaker-debugger/tensorflow2/tensorflow2_keras_custom_container/tf2-keras-custom-container.ipynb

```diff
@@ -0,0 +1,319 @@
+   "execution_count": 4,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# define the entrypoint script\n",
+    "entrypoint_script='src/tf_keras_resnet_byoc.py'"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Setting up the Estimator\n",
+    "\n",
+    "Now it's time to setup our TensorFlow estimator. We've added new parameters to the estimator to enable your training job for debugging through Amazon SageMaker Debugger. These new parameters are explained below.\n",
+    "\n",
+    "* **debugger_hook_config**: This new parameter accepts a local path where you wish your tensors to be written to and also accepts the S3 URI where you wish your tensors to be uploaded to. SageMaker will take care of uploading these tensors transparently during execution.\n",
+    "* **rules**: This new parameter will accept a list of rules you wish to evaluate against the tensors output by this training job. For rules, Amazon SageMaker Debugger supports two types:\n",
+    " * **SageMaker Rules**: These are rules specially curated by the data science and engineering teams in Amazon SageMaker which you can opt to evaluate against your training job.\n",
+    " * **Custom Rules**: You can optionally choose to write your own rule as a Python source file and have it evaluated against your training job. To provide Amazon SageMaker Debugger to evaluate this rule, you would have to provide the S3 location of the rule source and the evaluator image.\n",
+    " \n",
+    "#### Using Amazon SageMaker Rules\n",
+    " \n",
+    "In this example we'll demonstrate how to use SageMaker rules to be evaluated against your training. You can find the list of SageMaker rules and the configurations best suited for using them [here](https://github.com/awslabs/sagemaker-debugger-rulesconfig).\n",
+    "\n",
+    "The rules we'll use are **VanishingGradient** and **LossNotDecreasing**. As the names suggest, the rules will attempt to evaluate if there are vanishing gradients in the tensors captured by the debugging hook during training and also if the loss is not decreasing."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "rules = [\n",
+    "    Rule.sagemaker(rule_configs.vanishing_gradient()), \n",
+    "    Rule.sagemaker(rule_configs.loss_not_decreasing())\n",
+    "]"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Let us now create the estimator and call `fit()` on our estimator to start the training job and rule evaluation job in parallel."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "estimator = TensorFlow(\n",
+    "    role=sagemaker.get_execution_role(),\n",
+    "    base_job_name='smdebug-demo-tf2-keras',\n",
+    "    ## custom container with TF 2.1.0 and smdebug 0.7.1 installed\n",
+    "    #image_name='',\n",
```

Comment:
may be more comment here and above, which tells that they have to insert their own image and link BYOC page. Actually information about BYOC should be at the top of notebook.

Reply:
added a link about BYOC at the top of the notebook